### ISSUE TITLE
"error: expected token: ':'; generated -> instead" on destructured lambda parameters with types

### ISSUE THREAD

**Issue by ZacSweers** [facebookincubator/ktfmt#323]
A snippet like this, while somewhat exotic, is legal in kotlin but breaks the formatter

```kotlin
      .doOnNext { (_, _, _, threads): ThreadsViewApiResponse ->
        // ...
      }
```

Where `ThreadsViewApiResponse` is a data class with four component functions that are destructured here. The above results in the following error. Removing the explicit type does work around the error, but I think it should be supported syntax

```
> Task :app:spotlessKotlin
Step 'ktfmt' found problem in 'src/androidTest/java/slack/app/ui/e2e/SignedInBaseTest.kt':
647:29: error: expected token: ':'; generated -> instead

com.google.googlejavaformat.FormattingError: 647:29: error: expected token: ':'; generated -> instead

        at com.google.googlejavaformat.OpsBuilder.token(OpsBuilder.java:315)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.token(KotlinInputAstVisitor.kt:2308)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.token$default(KotlinInputAstVisitor.kt:2307)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitLambdaExpression(KotlinInputAstVisitor.kt:833)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitLambdaExpression(KotlinInputAstVisitor.kt:783)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitLambdaExpression(KtVisitorVoid.java:795)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitLambdaExpression(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtLambdaExpression.accept(KtLambdaExpression.java:40)
        at org.jetbrains.kotlin.psi.KtLambdaExpression.accept(KtLambdaExpression.java:92)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visit(KotlinInputAstVisitor.kt:2355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitArgument(KotlinInputAstVisitor.kt:1004)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitArgument(KtVisitorVoid.java:657)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitArgument(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtValueArgument.accept(KtValueArgument.java:46)
        at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:49)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visit(KotlinInputAstVisitor.kt:2355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitCallElement(KotlinInputAstVisitor.kt:753)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.emitQualifiedExpression(KotlinInputAstVisitor.kt:526)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitQualifiedExpression(KotlinInputAstVisitor.kt:467)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitQualifiedExpression(KtVisitorVoid.java:819)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitQualifiedExpression(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtVisitor.visitDotQualifiedExpression(KtVisitor.java:302)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDotQualifiedExpression(KtVisitorVoid.java:289)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDotQualifiedExpression(KtVisitorVoid.java:843)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDotQualifiedExpression(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtDotQualifiedExpression.accept(KtDotQualifiedExpression.kt:32)
        at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:49)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visit(KotlinInputAstVisitor.kt:2355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitStatement(KotlinInputAstVisitor.kt:401)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitStatements(KotlinInputAstVisitor.kt:414)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitBlockBody(KotlinInputAstVisitor.kt:390)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitFunctionLikeExpression(KotlinInputAstVisitor.kt:355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitNamedFunction(KotlinInputAstVisitor.kt:162)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedFunction(KtVisitorVoid.java:487)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedFunction(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtNamedFunction.accept(KtNamedFunction.java:51)
        at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:49)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visit(KotlinInputAstVisitor.kt:2355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitStatement(KotlinInputAstVisitor.kt:401)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitStatements(KotlinInputAstVisitor.kt:414)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitBlockBody(KotlinInputAstVisitor.kt:390)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitClassOrObject(KotlinInputAstVisitor.kt:1327)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassOrObject(KtVisitorVoid.java:469)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassOrObject(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtVisitor.visitClass(KtVisitor.java:33)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClass(KtVisitorVoid.java:33)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClass(KtVisitorVoid.java:463)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClass(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtClass.accept(KtClass.kt:20)
        at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:49)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visit(KotlinInputAstVisitor.kt:2355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitScript(KotlinInputAstVisitor.kt:2276)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitScript(KtVisitorVoid.java:523)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitScript(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtScript.accept(KtScript.java:69)
        at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:49)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visit(KotlinInputAstVisitor.kt:2355)
        at com.facebook.ktfmt.format.KotlinInputAstVisitor.visitKtFile(KotlinInputAstVisitor.kt:2254)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:517)
        at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:21)
        at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.kt:245)
        at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.kt:232)
        at com.facebook.ktfmt.format.Formatter.prettyPrint(Formatter.kt:102)
        at com.facebook.ktfmt.format.Formatter.format(Formatter.kt:85)
```

**Comment by cgrushko** [comment]
Thanks for the report! I think this is fixed at head. I'm running some last checks and will push a new version to github.

**Comment by cgrushko** [comment]
Verified this works on released v0.38, lemme know if you see otherwise.
